Add support for Alpha 17 C7VF

### Laptop model

Alpha 17 C7VF

### EC firmware version

17KKIMS1.114

### List of addresses and their values

```text
0000  00 80 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
0010  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
0020  00 00 00 00 00 00 00 00  0a 05 00 00 00 00 4b 52  |..............KR|
0030  03 09 00 0d 01 00 50 81  6a 18 60 3b 71 02 e0 00  |......P.j.`;q...|
0040  00 00 64 00 0b 18 00 00  0b 18 2c 43 c3 0b 00 00  |..d.......,C....|
0050  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
0060  00 00 00 00 00 00 00 00  3f 00 32 39 40 47 4e 55  |........?.29@GNU|
0070  6c 32 00 28 2d 32 39 41  50 69 04 04 04 04 04 09  |l2.(-29APi......|
0080  2a 00 37 3c 41 46 4b 50  62 00 00 30 37 3e 44 4d  |*.7<AFKPb..07>DM|
0090  55 64 04 04 04 04 04 04  06 00 00 06 00 00 00 00  |Ud..............|
00a0  31 37 4b 4b 49 4d 53 31  2e 31 31 34 31 30 31 37  |17KKIMS1.1141017|
00b0  32 30 32 34 31 35 3a 30  31 3a 33 34 00 00 00 08  |202415:01:34....|
00c0  00 00 06 22 00 00 00 00  00 ad 00 aa 00 00 00 00  |..."............|
00d0  00 00 c4 00 0d 00 03 bc  00 01 00 00 00 00 00 00  |................|
00e0  e2 00 00 0b 18 00 00 40  01 00 00 00 00 c0 00 00  |.......@........|
00f0  00 00 70 00 2e 20 2e 80  28 50 00 00 00 03 00 00  |..p.. ..(P......|

battery:
0xd7
E4 // best for mobility
D0 //balanced(80%)
BC //best for mobility(60%)

fan1 curve:
from 0x72 to 0x77 //percentage
from 0x69 to 0x6E //probably temperature, not sure

fan2 curve:
from 0x8A to 0x8F
from 0x81 to 0x86

fan mode:
0x98
06 //advanced
86 //cooler boost
06 //auto, with default value 00 28 2D 32 39 41 at 0x72 to 0x77 and 00 30 37 3E 44 4D at 0x8A to 0x8F

usb power share:
0xbf
28 //on
o8 //off
```

### GPU

Nvidia

### Is your keyboard RGB?

Yes (multi color RGB)

### Additional context

I only find these values. If more are needed, please tell me. And FORGIVE me for my poooooor English.

$ sudo make dkms-install
dkms --version >> /dev/null
mkdir -p /usr/src/msi_ec-0.08
cp /home/lhx/Downloads/msi-ec/dkms.conf /usr/src/msi_ec-0.08
cp /home/lhx/Downloads/msi-ec/Makefile /usr/src/msi_ec-0.08
cp /home/lhx/Downloads/msi-ec/msi-ec.c /usr/src/msi_ec-0.08
cp /home/lhx/Downloads/msi-ec/ec_memory_configuration.h /usr/src/msi_ec-0.08
sed -e "s/@CFLGS@//" \
    -e "s/@VERSION@/0.08/" \
    -i /usr/src/msi_ec-0.08/dkms.conf
dkms add msi_ec/0.08
Error! DKMS tree already contains: msi_ec-0.08
You cannot add the same module/version combo more than once.
make: *** [Makefile:52: dkms-install] Error 3

--- a/msi-ec.c
+++ b/msi-ec.c
@@ -3303,6 +3303,78 @@ static struct msi_ec_conf CONF46 __initdata = {
 	},
 };
 
+static const char *ALLOWED_FW_47[] __initconst = {
+	"17KKIMS1.108", // Alpha 17 C7VF / C7VG
+	"17KKIMS1.109",
+	"17KKIMS1.114",
+	NULL
+};
+
+static struct msi_ec_conf CONF47 __initdata = {
+	.allowed_fw = ALLOWED_FW_47, // WMI2 based
+	.charge_control_address = 0xd7,
+	// .usb_share = {
+	//  	.address      = 0xbf, // states: 0x08 || 0x28
+	//  	.bit          = 5,
+	// },
+	.webcam = {
+		.address       = 0x2e,
+		.block_address = MSI_EC_ADDR_UNSUPP,
+		.bit           = 1,
+	},
+	.fn_win_swap = {
+		.address = 0xe8,
+		.bit     = 4,
+		.invert  = true,
+	},
+	.cooler_boost = {
+		.address = 0x98,
+		.bit     = 7,
+	},
+	.shift_mode = {
+		.address = 0xd2,
+		.modes = {
+			{ SM_ECO_NAME,     0xc2 }, // eco-silent
+			{ SM_COMFORT_NAME, 0xc1 }, // balanced
+			{ SM_TURBO_NAME,   0xc4 }, // extreme
+			MSI_EC_MODE_NULL
+		},
+	},
+	.super_battery = {
+		.address = MSI_EC_ADDR_UNSUPP, // 0xEB=0 unchanged
+		.mask    = 0x0f,
+	},
+	.fan_mode = {
+		.address = 0xd4,
+		.modes = {
+			{ FM_AUTO_NAME,     0x0d }, // auto, cooler boost, balanced, eco-silent
+			{ FM_ADVANCED_NAME, 0x8d }, // advanced
+			MSI_EC_MODE_NULL
+		},
+	},
+	.cpu = {
+		.rt_temp_address      = 0x68,
+		.rt_fan_speed_address = 0x71,
+	},
+	.gpu = {
+		.rt_temp_address      = 0x80,
+		.rt_fan_speed_address = 0x89,
+	},
+	.leds = {
+		.micmute_led_address = MSI_EC_ADDR_UNSUPP,
+		.mute_led_address    = MSI_EC_ADDR_UNSUPP,
+		.bit                 = 1,
+	},
+	.kbd_bl = {
+		.bl_mode_address  = MSI_EC_ADDR_UNSUPP,
+		.bl_modes         = { 0x00, 0x08 },
+		.max_mode         = 1,
+		.bl_state_address = MSI_EC_ADDR_UNSUPP,
+		.state_base_value = 0x80,
+		.max_state        = 3,
+	},
+};
+
 static struct msi_ec_conf *CONFIGURATIONS[] __initdata = {
 	&CONF0,
 	&CONF1,
@@ -3351,6 +3423,7 @@ static struct msi_ec_conf *CONFIGURATIONS[] __initdata = {
 	&CONF44,
 	&CONF45,
 	&CONF46,
+	&CONF47,
 	NULL
 };
 

